### PR TITLE
Add Code Climate integration

### PR DESCRIPTION
**Pull request by DeeDeeG**
# Context
- Adds Code Climate integration and badges

# Summary of Changes

- Adds a few code climate test reporting integration steps to the Travis CI config
- Require `simplecov` at the beginning of the Cucumber `env.rb` file
- Adds two badges to README.md

# Checklist

- [ ] Tested Mobile Responsiveness
- [ ] Added Unit Tests
- [x] CI Passes
- [ ] Deploys to Heroku on test Correctly (Maintainers will handle)
- [ ] Added Documentation (Service and Code when required)

# Screenshots

## Before
readme.md-before

## After

readme.md-after-showing-working-status-badges

### CODE REVIEW

**Comment by DeeDeeG** [comment]
I guess this needs another commit to run `simplecov` somewhere in the test steps

**Comment by DeeDeeG** [comment]
We get this in the Travis log:

```
"failed to read file /refugerestrooms/app/controllers/restrooms_controller.rb\nopen /refugerestrooms/app/controllers/restrooms_controller.rb: no such file or directory"
```

```
$ ./cc-test-reporter after-build --coverage-input-type simplecov --exit-code $TRAVIS_TEST_RESULT
time="2018-02-09T21:10:18Z" level=error msg="failed to read file /refugerestrooms/app/controllers/restrooms_controller.rb\nopen /refugerestrooms/app/controllers/restrooms_controller.rb: no such file or directory" 
Error: open /refugerestrooms/app/controllers/restrooms_controller.rb: no such file or directory
Usage:
  cc-test-reporter after-build [flags]
Flags:
  -s, --batch-size int               batch size for source files (default 500)
  -e, --coverage-endpoint string     endpoint to upload coverage information to (default "https://api.codeclimate.com/v1/test_reports")
  -t, --coverage-input-type string   type of input source to use [clover, cobertura, coverage.py, excoveralls, gcov, gocov, jacoco, lcov, simplecov]
      --exit-code int                exit code of the test run
  -r, --id string                    reporter identifier (default "c18df080592f9c99ca8080a6d5e052aa5fd3964044a0fe0b71e48f8e18998dc2")
  -p, --prefix string                the root directory where the coverage analysis was performed (default "/home/travis/build/RefugeRestrooms/refugerestrooms")
Global Flags:
  -d, --debug   run in debug mode
```

Line 5261: https://travis-ci.org/RefugeRestrooms/refugerestrooms/builds/339645924#L5261

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,16 @@
...
 before_install:
   - docker-compose build
+before_script:
+  - curl -L https://codeclimate.com/downloads/test-reporter/test-reporter-0.1.4-linux-amd64 > ./cc-test-reporter
+  - chmod +x ./cc-test-reporter
+  - ./cc-test-reporter before-build
 script:
   - docker-compose run -e "RAILS_ENV=test" web rake db:test:prepare spec cucumber
+after_script:
+  - ./cc-test-reporter after-build -p "/home/travis/build/RefugeRestrooms/refugerestrooms" --coverage-input-type simplecov --exit-code $TRAVIS_TEST_RESULT
```
Comment (DeeDeeG) [inline .travis.yml]: Trying to edit this line and see if I can get it to work, since it errors out in Travis CI

**Comment by DeeDeeG** [comment]
Checking whether this is a problem accessing files when using Docker in Travis

**Comment by tkwidmer** [comment]
This looks good to me.

**Comment by DeeDeeG** [comment]
We are still getting a "?" badge for code coverage. I am still not quite sure how that works.

This is an outline of how it's _supposed_ to work:

- SimpleCov analyzes our Ruby code, outputs "coverage report" file(s)

- There is a small code climate "reporter" executable we download, which I believe just sends those SimpleCov reports to codeclimate.com

What's apparently going wrong:

- We get this error when trying to run code climate's reporter executable:

```
$ ./cc-test-reporter after-build --coverage-input-type simplecov --exit-code $TRAVIS_TEST_RESULT
time="2018-02-09T21:10:18Z" level=error msg="failed to read file /refugerestrooms/app/controllers/restrooms_controller.rb\nopen /refugerestrooms/app/controllers/restrooms_controller.rb: no such file or directory" 
Error: open /refugerestrooms/app/controllers/restrooms_controller.rb: no such file or directory
Usage:
  cc-test-reporter after-build [flags]
```

**Comment by stardust66** [comment]
@DeeDeeG, I think you need to run the `after-build` command with `--prefix /refugerestrooms`. It says in the Code Climate documentation here: https://docs.codeclimate.com/docs/test-coverage-troubleshooting-tips

> You may need to specify a prefix value if your tests are run outside of your project's root directory.
> For example, if you're running tests within a docker container and your app code is located at `/usr/src/app`, you'd use:
> ```
> cc-test-reporter-format-coverage --prefix /usr/src/app
> ```

It's not clear that we need this, because I don't think we're running tests outside the project's root directory, but it mentioned Docker so it might be worth trying.

**Comment by DeeDeeG** [comment]
This is totally working, thanks to the research and PR by @stardust66. 

Check out the [`code-climate`](https://github.com/RefugeRestrooms/refugerestrooms/tree/code-climate#readme) branch for a live preview of how this will look. (I'd call this done and ready to merge.)

**Comment by DeeDeeG** [comment]
These merge conflicts are pretty silly, and I don't think we need 6 commits total in this PR in order to deal with them.

If no-one minds, I have a rebased branch that I can overwrite this branch with. See this rich diff:  https://github.com/RefugeRestrooms/refugerestrooms/compare/develop...DeeDeeG:code-climate-rebased-on-develop

**Comment by DeeDeeG** [comment]
Pushed the rebased branch. 

Old branch is at [`code-climate-old`](https://github.com/RefugeRestrooms/refugerestrooms/tree/code-climate-old) for now, but will probably delete after this is merged.

**Comment by DeeDeeG** [comment]
Since this got plenty of approval at the time, even before it was working, I'm just going to assume this is approved for merge.

Intending to merge tomorrow if no-one has any objections.

**Comment by tkwidmer** [comment]
LGTM still

**Pull request merged**